subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationStep.java

```diff
@@ -65,7 +66,7 @@ public boolean hasCachedResult(TransformationSubject subject) {
             return true;
         }
         for (File file : subject.getFiles()) {
-            if (!transformerInvoker.hasCachedResult(file, transformer)) {
+            if (!transformerInvoker.contains(file.getAbsoluteFile(), transformer.getSecondaryInputHash())) {
```

Comment:
The original code read much better than this.

Reply:
The API of transformer invoker now is back to what it was before.